### PR TITLE
ENG-612: Set token.origin in synchronizeTokensFromData

### PR DESCRIPTION
**Pull request by tchief**
### Changes
- Set `token.origin` in synchronizeTokensFromData

### CODE REVIEW

File: src/model/support/SDKSource.ts [hunk 1]
```diff
@@ -12,8 +12,10 @@
 // --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---
 // MARK: - Definitions
 
+export type SourceType = "Figma" | "TokenStudio"
```
Comment (JiriTrecak) [inline src/model/support/SDKSource.ts:15]: Can you please change it to enum? figma="Figma", tokenStudio="TokenStudio", entire SDK uses enums almost exclusively

File: src/tools/design-tokens/SDKToolsDesignTokensPlugin.ts [hunk 2]
```diff
@@ -269,6 +270,21 @@ export class SupernovaToolsDesignTokensPlugin {
...
   }
 
+  private setTokensOrigin(mapping: DTPluginToSupernovaMapPack): DTPluginToSupernovaMapPack {
+    const sourceId = this.version.designSystem.sources.find(s => s.type === "TokenStudio")?.id
+    for (let map of mapping) {
+      for (let node of map.processedNodes) {
+        node.token.origin = {
+          name: node.token.name,
+          sourceId,
+          id: node.key.toLowerCase(),
```
Comment (JiriTrecak) [inline src/tools/design-tokens/SDKToolsDesignTokensPlugin.ts:280]: In order to do this, we'll need to lowercase the key on creation of it, so the key is the same everywhere. This way, it just reports on the key as lowercase, but computation will still think two equal key with different cases will be different. There is `buildKey` function somewhere, I believe

**Review by JiriTrecak — APPROVED** [review]

**Comment by JiriTrecak** [comment]
@tchief I approved the PR but please make changes based on the comment I left

File: src/tools/design-tokens/SDKToolsDesignTokensPlugin.ts (hunk 2 above)
Comment (tchief) [reply src/tools/design-tokens/SDKToolsDesignTokensPlugin.ts]: @JiriTrecak Done

Comment (JiriTrecak) [reply src/tools/design-tokens/SDKToolsDesignTokensPlugin.ts]: Tyty :))

**Comment by JiriTrecak** [comment]
@tchief feel free to merge?

**Pull request merged**